rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java

```diff
@@ -147,4 +163,40 @@ private String getLoggedInUserPreferredName() {
         final AllSharedPreferences allSharedPreference = RDTApplication.getInstance().getContext().allSharedPreferences();
         return allSharedPreference.getANMPreferredName(allSharedPreference.fetchRegisteredANM());
     }
+
+    private JSONArray getLocations() throws JSONException {
+        JSONArray jsonArray = new JSONArray();
+        AllSharedPreferences allSharedPreferences = RDTApplication.getInstance().getContext().allSharedPreferences();
+        String defaultLocationUuid = allSharedPreferences.fetchDefaultLocalityId(allSharedPreferences.fetchRegisteredANM());
+        LinkedHashMap<String, TreeNode<String, Location>> locationMap = LocationHelper.getInstance().map();
```

Comment:
Isn't this starting from the root location? The reason for enabling `RDTSyncAllLocationsIntentService` is to be able to start traversal from the immediate parent location.

Reply:
it's starting from the root location by using `map()` and traverse to the immediate parent location by the given child location id in a param and get all the child locations from the immediate parent by the given child location id.

Reply:
As discussed, after enabling `SyncAllLocationsServiceJob` you should be able to get the immediate parent hierarchy from the `LocationRepository`.